src/main/java/ch/epfl/biop/operetta/OperettaManager.java

```diff
@@ -906,9 +986,67 @@ public void process(List<Well> wells, List<Integer> fields, Roi region) {
+                                    break;
+                                case (ImagePlus.GRAY16):
+                                    pixelType = PixelType.UINT16;
+                                    break;
+                                case (ImagePlus.GRAY32):
+                                    pixelType = PixelType.FLOAT;
+                                    break;
+                                default:
+                                    throw new IllegalArgumentException("Unknown ImagePlus type " + well_image.getType());
+                            }
+
+                            Calibration cal = well_image.getCalibration();
+
+                            // create the well object
+                            WellCompanion wellCompanion = new WellCompanion.Builder()
+                                    .setRow(well.getRow().getValue())
+                                    .setColumn(well.getColumn().getValue())
+                                    .build();
+                            String wellId = companionFileGenerator.addWell(wellCompanion);
+
+                            // get and set the current instrument
+                            Instrument instrument = getInstrument(0);
+                            String instrumentId = companionFileGenerator.setInstrument(instrument);
+
+                            // create the image object
+                            int serieId = well_fields.get(0).getIndex().getValue();
+                            ImageCompanion imageCompanion = new ImageCompanion.Builder()
+                                    .setName(name + ".ome.tiff")
+                                    .addGlobalMetadata(globalMetadataMap)
+                                    .setPixelSizeX(new Length(cal.pixelWidth, UNITS.MICROMETER))
+                                    .setPixelSizeY(new Length(cal.pixelHeight, UNITS.MICROMETER))
+                                    .setDimensionOrder(getDimensionOrder(serieId))
+                                    .setPixelType(pixelType)
+                                    .setSizeC(well_image.getNChannels())
+                                    .addChannels(getChannels(serieId, this.range.getRangeC()))
+                                    .setSizeT(well_image.getNFrames())
+                                    .setSizeZ(well_image.getNSlices())
+                                    .setSizeY(well_image.getHeight())
+                                    .setSizeX(well_image.getWidth())
+                                    .setObjectiveSettings(getObjectiveSettings(serieId))
+                                    .setInstrument(instrument)
+                                    .setAcquisitionDate(getAcquisitionDate(serieId))
+                                    .build();
+                            companionFileGenerator.addImage(imageCompanion, wellId, plateAcquisitionId);
+                        }
+                        else
```

Comment:
Please add { and }, even with one-liners.

Reply:
Done